server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -88,7 +90,12 @@ public int getShardLimitPerNode() {
      * @param state          the current cluster state
      * @throws ValidationException if creating this index would put the cluster over the cluster shard limit
      */
-    public void validateShardLimit(final Settings settings, final ClusterState state) {
+    public void validateShardLimit(final String indexName, final Settings settings, final ClusterState state) {
+        // Validate shard limit only for non system indices as it is not hard limit anyways
```

Comment:
Does it harm the system to relax the limit for certain types? What are the consequences?

Reply:
This limit is soft to prevent resources going overboard. It is fine to prevent creation or opening of data indices as the impact is constrained to individual index, but doing that for system index can make feature completely unusable. Also, system indices are generally lightweight and well provisioned, hence should not be constrained by this soft limit.